# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RenderCompositor.cpp
# Commit: 99b2e5c55a25
# Full Hash: 99b2e5c55a25f47f9a5a0650607cd54ce966c28b
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2020-11-05 16:09:03
# Regressor Bug: 1673983
# File Overlap Count: 2
# Description:
#   Bug 1673983 - Add D3D11 RenderCompositor for SWGL. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D95797
# ==============================================================================

diff -r 96087b511fe9 -r 99b2e5c55a25 gfx/webrender_bindings/RenderCompositor.cpp
--- a/gfx/webrender_bindings/RenderCompositor.cpp	Thu Nov 05 00:51:24 2020 +0000
+++ b/gfx/webrender_bindings/RenderCompositor.cpp	Thu Nov 05 09:14:52 2020 +0000
@@ -6,7 +6,9 @@
 
 #include "RenderCompositor.h"
 
+#include "gfxConfig.h"
 #include "GLContext.h"
+#include "mozilla/StaticPrefs_gfx.h"
 #include "mozilla/gfx/gfxVars.h"
 #include "mozilla/layers/SyncObject.h"
 #include "mozilla/webrender/RenderCompositorOGL.h"
@@ -15,6 +17,7 @@
 
 #ifdef XP_WIN
 #  include "mozilla/webrender/RenderCompositorANGLE.h"
+#  include "mozilla/webrender/RenderCompositorD3D11SWGL.h"
 #endif
 
 #if defined(MOZ_WAYLAND) || defined(MOZ_WIDGET_ANDROID)
@@ -138,9 +141,17 @@
 #ifdef XP_MACOSX
     // Mac uses NativeLayerCA
     return RenderCompositorNativeSWGL::Create(std::move(aWidget), aError);
-#else
+#elif defined(XP_WIN)
+    if (StaticPrefs::gfx_webrender_software_d3d11_AtStartup() &&
+        gfx::gfxConfig::IsEnabled(gfx::Feature::D3D11_COMPOSITING)) {
+      UniquePtr<RenderCompositor> comp =
+          RenderCompositorD3D11SWGL::Create(std::move(aWidget), aError);
+      if (comp) {
+        return comp;
+      }
+    }
+#endif
     return RenderCompositorSWGL::Create(std::move(aWidget), aError);
-#endif
   }
 
 #ifdef XP_WIN